credsweeper/file_handler/patch_provider.py

```diff
@@ -0,0 +1,57 @@
+from typing import Dict, List, Optional
+
+from credsweeper.file_handler.files_provider import FilesProvider
+from credsweeper.file_handler.diff_content_provider import DiffContentProvider
+from credsweeper.utils import Util
+
+
+class PatchProvider(FilesProvider):
+    """Provider of patch files class
```

Comment:
```suggestion
    """Provide data from a list of `.patch` files. 
    
    Allows to scan for data that has changed between git commits, rather than the entire project.
```